File: doc-examples/example-protobuf-java/src/test/java/example/Temperature.java
package example;

import io.micronaut.serde.annotation.Serdeable;
import io.micronaut.serde.protobuf.annotation.ProtoField;
import io.micronaut.serde.protobuf.annotation.ProtoType;

/**
 * The same value under the two representations protobuf offers for a signed 32-bit integer.
 */
public final class Temperature {

    private Temperature() {
    }

    // tag::clazz[]
    @Serdeable
    public record AsInt32(@ProtoField(1) int milliCelsius) {
    }

    @Serdeable
    public record AsSint32(@ProtoField(value = 1, type = ProtoType.SINT32) int milliCelsius) {
    }
    // end::clazz[]
}

File: doc-examples/example-protobuf-kotlin/src/test/kotlin/example/Temperature.kt
package example

import io.micronaut.serde.annotation.Serdeable
import io.micronaut.serde.protobuf.annotation.ProtoField
import io.micronaut.serde.protobuf.annotation.ProtoType

/**
 * The same value under the two representations protobuf offers for a signed 32-bit integer.
 */
class Temperature private constructor() {

    // tag::clazz[]
    @Serdeable
    data class AsInt32(@field:ProtoField(1) val milliCelsius: Int)

    @Serdeable
    data class AsSint32(@field:ProtoField(value = 1, type = ProtoType.SINT32) val milliCelsius: Int)
    // end::clazz[]
}
